src/pharmpy/model/external/nonmem/parsing.py

```diff
@@ -373,7 +374,31 @@ def parse_estimation_steps(control_stream, random_variables) -> EstimationSteps:
     etaderiv_names = None
     epsilonderivs_names = None
     table_records = control_stream.get_records('TABLE')
+    predictions = ()
+    residuals = ()
+    all_predictions = ['PRED', 'CPRED', 'CPREDI', 'CIPRED', 'CIPREDI', 'IPRED']
+    all_residuals = [
+        'RES',
+        'WRES',
+        'CRES',
+        'CWRES',
+        'CRESI',
+        'CWRESI',
+        'CIRES',
+        'CIWRES',
+        'CIRESI',
+        'CIWRESI',
+    ]
+    options_match = (
+        r"\S+=\S+\s*|\S*(?:PRINT|APPEND|ONLY|HEADER"
+        r"|FORWARD|CONDITIONAL|TITLE|LABEL|OMITTED|WRESCHOL)\S*\s*"
+    )
```

Comment:
It is rather complex to parse $TABLE. It is done in `nonmem.records.table_record.parse_options`. It is used in `parse_table_columns`. I think it should be used here.